File: support/src/main/java/com/wuguangxin/adapter/BaseFragmentAdapter.java
package com.wuguangxin.adapter;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * Fragment适配器
 *
 * Created by wuguangxin on 17/3/27
 */
public class BaseFragmentAdapter extends FragmentPagerAdapter {
	private List<? extends Fragment> list;
	private List<String> titleList = new ArrayList<>();

	public BaseFragmentAdapter(FragmentActivity activity) {
		super(activity.getSupportFragmentManager());
	}

	public BaseFragmentAdapter(FragmentManager fm) {
		super(fm);
	}

	public BaseFragmentAdapter(FragmentManager fm, List<? extends Fragment> list) {
		super(fm);
		this.list = list;
	}

	public BaseFragmentAdapter(FragmentManager fm, List<? extends Fragment> list, List<String> titleList) {
		super(fm);
		this.list = list;
		this.titleList = titleList;
	}

	@Override
	public Fragment getItem(int position) {
		return list == null ? null : list.get(position);
	}

	@Override
	public int getCount() {
		return list == null ? 0 : list.size();
	}

	@Nullable
	@Override
	public CharSequence getPageTitle(int position) {
		return titleList == null ? null : titleList.get(position);
	}

	public void setList(List<? extends Fragment> list) {
		this.list = list;
		notifyDataSetChanged();
	}

	public void setList(List<? extends Fragment> fragmentList, List<String> titleList) {
		this.list = fragmentList;
		this.titleList = titleList;
		notifyDataSetChanged();
	}

	public void setTitleList(List<String> titleList) {
		this.titleList = titleList;
		notifyDataSetChanged();
	}
}

File: support/src/main/java/com/wuguangxin/adapter/BaseFragmentAdapter.kt
package com.wuguangxin.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentPagerAdapter
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentManager
import java.util.ArrayList

/**
 * Fragment适配器
 *
 * Created by wuguangxin on 17/3/27
 */
class BaseFragmentAdapter : FragmentPagerAdapter {
    private val mFragments = arrayListOf<Fragment>()
    private val mTitles = mutableListOf<String>()

    constructor(activity: FragmentActivity) : this(activity.supportFragmentManager)
    constructor(fm: FragmentManager) : this(fm, null)
    constructor(fm: FragmentManager, fragments: MutableList<Fragment>?) : this(fm, fragments, null)
    constructor(fm: FragmentManager, fragments: MutableList<Fragment>?, titles: MutableList<String>?): super(fm) {
        setList(fragments, titles)
    }

    override fun getItem(position: Int): Fragment {
        return mFragments[position]
    }

    override fun getCount(): Int {
        return mFragments.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return mTitles[position]
    }

    fun setTitleList(titles: MutableList<String>?) {
        this.mTitles.clear()
        titles?.let { this.mTitles.addAll(titles) }
        notifyDataSetChanged()
    }

    fun setList(fragments: MutableList<Fragment>?) {
        this.mFragments.clear()
        fragments?.let { this.mFragments.addAll(fragments) }
        notifyDataSetChanged()
    }

    fun setList(fragments: MutableList<Fragment>?, titles: MutableList<String>?) {
        this.mFragments.clear()
        this.mTitles.clear()
        fragments?.let { this.mFragments.addAll(fragments) }
        titles?.let { this.mTitles.addAll(titles) }
        notifyDataSetChanged()
    }
}